lib/puppet/provider/foreman_resource/rest_v3.rb

```diff
@@ -55,6 +55,10 @@ def generate_token
     OAuth::AccessToken.new(oauth_consumer)
   end
 
+  def fqdn
+    @fqdn ||= URI::parse(resource[:base_url]).host
+  end
```

Comment:
You could simplify it and leave this method out. Then in `error_message` you can locally define `fqdn = URI::parse(resource[:base_url]).host`.

Reply:
True!

Reply:
Cool, this way we call `URI::parse` only once and we avoid the extra method. I'll make the change.